README.md

```diff
@@ -1 +1,112 @@
-# Room Access rules
\ No newline at end of file
+# Room Access rules
+
+This module implements handling around the `im.vector.room.access_rules` state event, which spec is described below.
+
+## `im.vector.room.access_rules`
+
+Restricts the access to a room based on the selected preset. Body:
+
+```json
+{
+    "rule": "<rule>"
+}
+```
+
+`<rule>` is either `restricted`, `unrestricted` or `direct`.
+
+The implementation of the different presets lives in the
+`synapse.third_party_rules.access_rules` module.
+
+### `restricted` preset
+
+Default preset for non-direct rooms.
+
+Forbids any invite and membership update for users that belong to a server
+that is in the blacklist provided by the server's configuration
+(`domains_forbidden_when_restricted`). If the invite is a 3PID invite, queries
+a custom `/info` endpoint of the configured identity server to check if that email
+address would belong to a blacklisted server.
+
+### `unrestricted` preset
+
+Doesn't apply any restriction on who can join the room.
+
+Forbids any `m.room.power_levels` event that either:
+
+* change the default power level to a non-0 value, or
+* change the power level for a user from a blacklisted server (see details about the `restricted` preset) to a non-default value
+
+### `direct` preset
```

Comment:
I understand the rules of this preset now, but is there a succinct explanation of why this preset is useful that we could add to the top here?

Reply:
heh, well, fair. I mention this in case some other customer comes along asking (at a high level) for a similar implementation, but we don't have a high-level overview of what this one does, so we don't think to re-use this implementation.

Reply:
I'm not sure it's worth it, since a) I think you can get a better gist of it by reading down the rest of that section (which isn't super long), b) it's pretty Tchap-specific and not very flexible, and c) there's a chance at least part of this module is replaced with ACLs or something similar in the future.